calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -0,0 +1,309 @@
+                                netns_file]))
+
+        # Reach into the netns and delete the docker-allocated interface.
+        print(check_output(['ip', 'netns', 'exec', pid,
+                            'ip', 'link', 'del', 'eth0']))
+
+        # Clean up after ourselves (don't want to leak netns files)
+        print(check_output(['rm', netns_file]))
+
+    def _configure_profile(self, endpoint):
+        """
+        Configure the calico profile for a pod.
+
+        Currently assumes one pod with each name.
+        """
+        calicoctl('profile', 'add', self.pod_name)
+        pod = self._get_pod_config()
+        profile_name = self.pod_name
+
+        self._apply_rules(profile_name, pod)
+
+        self._apply_tags(profile_name, pod)
+
+        # Also add the workload to the profile.
+        calicoctl('endpoint', endpoint.endpoint_id,
+                  'profile', 'set', profile_name)
+        print('Finished configuring profile.')
+
+    def _get_pod_ports(self, pod):
+        """
+        Get the list of ports on containers in the Pod.
+
+        :return list ports: the Kubernetes ContainerPort objects for the pod.
+        """
+        ports = []
+        for container in pod['spec']['containers']:
+            try:
+                more_ports = container['ports']
+                print('Adding ports %s' % more_ports)
+                ports.extend(more_ports)
+            except KeyError:
+                pass
+        return ports
+
+    def _get_pod_config(self):
+        pods = self._get_api_path('pods')
+        print('Got pods %s' % pods)
+
+        for pod in pods:
+            print('Processing pod %s' % pod)
+            if pod['metadata']['name'].replace('-', '_') == self.pod_name:
+                this_pod = pod
+                break
+        else:
+            raise KeyError('Pod not found: ' + self.pod_name)
+        print('Got pod data %s' % this_pod)
+        return this_pod
+
+    def _get_api_path(self, path):
+        """
+        Get the list of pods from the Kube API server.
+
+        e.g.
+        _get_api_path('pods')
+
+        :param path: The relative path to an API endpoint.
+        :return: A list of JSON API objects
+        :rtype list
+        """
+        bearer_token = self._get_api_token()
+        session = requests.Session()
+        session.headers.update({'Authorization': 'Bearer ' + bearer_token})
+        response = session.get(
+            'https://kubernetes-master:6443/api/v1beta3/' + path,
+            verify=False,
+        )
+        response_body = response.text
+        # The response body contains some metadata, and the pods themselves
+        # under the 'items' key.
+        return json.loads(response_body)['items']
+
+    def _get_api_token(self):
+        """
+        Get the kubelet Bearer token for this node, used for HTTPS auth.
+        :return: The token.
+        :rtype: str
+        """
+        with open('/var/lib/kubelet/kubernetes_auth') as f:
```

Comment:
Hm, interesting.  I'll have to figure out why this is working for me without the file.  For now, we can leave as it and I will make sure my demo uses this authentication file.

We can track supporting different types of auth on the backlog.